packages/mobile/src/identity/verification.ts

```diff
@@ -222,61 +248,106 @@ export function* doVerificationFlow(withoutRevealing: boolean = false) {
+        ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_complete)
 
-      receiveMessageTask.cancel()
-      if (Platform.OS === 'android' && autoRetrievalTask) {
-        autoRetrievalTask.cancel()
+        // count how much more attestations we need to request
+        const attestationsToRequest =
+          status.numAttestationsRemaining - reveals.filter((r: boolean) => r).length
+
+        // check if we hit the limit for max actionable attestations at the same time
+        if (attestationsToRequest + attestations.length > MAX_ACTIONABLE_ATTESTATIONS) {
+          throw new Error(ErrorMessages.MAX_ACTIONABLE_ATTESTATIONS_EXCEEDED)
+        }
+
+        if (attestationsToRequest) {
+          yield put(setVerificationStatus(VerificationStatus.RequestingAttestations))
+          // request more attestations
+          ValoraAnalytics.track(VerificationEvents.verification_request_all_attestations_start, {
+            attestationsToRequest,
+          })
+          attestations = yield call(
+            requestAndRetrieveAttestations,
+            attestationsWrapper,
+            phoneHash,
+            account,
+            attestations,
+            attestations.length + attestationsToRequest
+          )
+          ValoraAnalytics.track(VerificationEvents.verification_request_all_attestations_complete, {
+            issuers,
+          })
+
+          // start listening for the new list of attestations
+          receiveMessageTask?.cancel()
+          issuers = attestations.map((a) => a.issuer)
+          receiveMessageTask = yield takeEvery(
+            Actions.RECEIVE_ATTESTATION_MESSAGE,
+            attestationCodeReceiver(attestationsWrapper, phoneHash, account, issuers)
+          )
+
+          ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_start)
+          // Request codes for the new list of attestations. We ignore unsuccessfull reveals here,
+          // cause we do not want to go into a loop of re-requesting more and more attestations
+          yield call(
+            revealAttestations,
+            attestationsWrapper,
+            account,
+            phoneHashDetails,
+            attestations
+          )
+          ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_complete)
+        }
+      }
+
+      yield put(setVerificationStatus(VerificationStatus.CompletingAttestations))
+      yield race({
+        actionableAttestationCompleted: all([
+          call(completeAttestations, attestationsWrapper, account, phoneHashDetails, attestations),
+          // Set acccount and data encryption key in Accounts contract
+          // This is done in other places too, intentionally keeping it for more coverage
+          call(registerAccountDek, account),
+        ]),
+        // This is needed, because we can have more actionableAttestations than NUM_ATTESTATIONS_REQUIRED
+        requiredAttestationsCompleted: call(function*() {
```

Comment:
Nit: I think ideally this generator would be split out into a helper rather than defined inline